File: src/main/java/org/jdc/template/json/DateTimeStringSerializer.java
package org.jdc.template.json;

import com.fasterxml.jackson.core.JsonGenerator;
import com.fasterxml.jackson.databind.JsonSerializer;
import com.fasterxml.jackson.databind.SerializerProvider;

import org.threeten.bp.LocalDateTime;
import org.threeten.bp.format.DateTimeFormatter;

import java.io.IOException;

public class DateTimeStringSerializer extends JsonSerializer<LocalDateTime> {
    public static final DateTimeFormatter FORMATTER = DateTimeFormatter.ISO_DATE_TIME;

    @Override
    public void serialize(LocalDateTime value, JsonGenerator jsonGenerator, SerializerProvider provider) throws IOException {
        if (value != null) {
            jsonGenerator.writeString(FORMATTER.format(value));
        } else {
            jsonGenerator.writeNull();
        }
    }
}

File: src/main/kotlin/org/jdc/template/json/DateTimeStringSerializer.kt
package org.jdc.template.json

import com.fasterxml.jackson.core.JsonGenerator
import com.fasterxml.jackson.databind.JsonSerializer
import com.fasterxml.jackson.databind.SerializerProvider

import org.threeten.bp.LocalDateTime
import org.threeten.bp.format.DateTimeFormatter

import java.io.IOException

class DateTimeStringSerializer : JsonSerializer<LocalDateTime>() {

    @Throws(IOException::class)
    override fun serialize(value: LocalDateTime?, jsonGenerator: JsonGenerator, provider: SerializerProvider) {
        if (value != null) {
            jsonGenerator.writeString(FORMATTER.format(value))
        } else {
            jsonGenerator.writeNull()
        }
    }

    companion object {
        val FORMATTER = DateTimeFormatter.ISO_DATE_TIME
    }
}
